src/integral.jl

```diff
@@ -50,6 +50,8 @@ function _integral(
         return _integral_gk_2d(f, geometry, rule; kwargs...)
     elseif N == 3
         return _integral_gk_3d(f, geometry, rule; kwargs...)
+    else
+        error("Integrating this geometry type with GaussKronrod not supported.")
```

Comment:
Could this become a more specific error type, like `MethodError` (I'm not sure if there exists a better error type)?

Reply:
Based on the docs, I'm not sure `MethodError` is the right choice here. I think `ArgumentError` might be the most appropriate, if we wanted to get away from from generic error types.

Reply:
I prefer specific error types. `ArgumentError` seems fine 👍

Reply:
I went through and changed out all instances of `error(msg)` with `throw`n specific errors.

Reply:
Thanks! Given the similarity between `_error_unsupported_gk` and `_error_unsupported_combination` would it be possible to fuse these two errors into one `_error_unsupported_combination`?

Reply:
I was looking at the cases where the generic GK-specific error was being called and realized that I was throwing errors on the 3D case and also for all >3D cases. I went ahead and eliminated the GK-specific error function and also consolidated the >=3D cases together.